## Урок 5 Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование

__Задача 1__

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

In [1]:
import numpy as np
from scipy import stats


### Решение:
__1. Формулируются нулевая и альтернативная гипотезы.__

1) Гипотеза $H_0$: средний диаметр шарика для подшипников равен 17.5 мм.

2) Гипотеза $H_1$: средний диаметр шарика для подшипников не равен 17.5 мм.

__2. Задаётся некоторая функция Т от выборки__

Предполагается, что количество шариков большое, то распределение диаметра шариков является нормальным или близким к нему. Т.к. случайная величина диаметра имеет нормальное распределение, то среднее значение повыборке $T$ будет также иметь нормальное распределение с параметрами $a = 17$ и $\sigma = \sqrt{\dfrac{D(X)}{n}} = \sqrt {\dfrac{4}{100}} = 0.2 $


In [2]:
a = 17
alpha = 0.025
m = 17.5
n = 100
D = 4
sigma = np.sqrt(D/n)
print('𝜎 =', np.sqrt(D/n))

samples = np.random.normal(17, 1, size=n)
print(samples)

𝜎 = 0.2
[17.19334958 17.00107482 19.09409858 16.96086608 17.12898986 16.16189279
 16.55436259 17.63112822 15.92343844 16.46308117 15.74192035 16.7779029
 16.99872744 15.87129938 16.74500563 17.63516316 18.44935564 16.95165088
 17.52042711 17.10678963 16.52780049 17.31948254 18.26781773 16.10939381
 16.89091293 18.54490487 16.99713678 16.96263515 17.66956954 17.41583052
 17.12587813 18.83459986 16.51304917 14.37762766 17.68965823 17.48011146
 16.07638923 15.96342432 18.08840034 17.65203181 17.80623629 17.18884025
 16.68372396 16.42426948 16.62807459 16.46976773 15.2928353  16.58282516
 16.02120642 16.91177991 16.27369298 16.19102623 16.97379361 16.01267806
 17.02898325 18.25657118 18.25523268 17.22731833 18.02843089 16.57680431
 18.55969177 17.75476934 16.39558609 18.10609574 16.23444973 18.0194712
 17.41876778 15.50102822 15.6435023  17.41911338 16.85802221 16.73960817
 17.56063932 16.8540865  18.48083872 16.35278377 18.46565386 16.61986834
 17.56515973 18.21007448 15.55745114 16.68525


__3. Фиксируется уровень значимости 𝛼__

    допустимая для данной задачи вероятность ошибки первого рода (чаще всего 0.01, 0.05 или 0.1).



In [3]:
alpha = 0.05

__4. Определяется критическая область $\Omega_\alpha = (t_{1 - \alpha}, \infty)$.__

Выбранный уровень значимости $\alpha = 0.025$ означает, что для проверки гипотезы о среднем значении a = 17 требуется, чтобы с вероятностью 95% выборочное среднее попадало в правосторонний интервал, который (по правилу 2-х сигм) равен $(a - 2 \cdot \sigma, a + 2 \cdot \sigma)$.

То есть критическая область лежит правее точки $a + 2 \cdot \sigma $ и если выборочное среднее = 17.5 попадёт в критическую область, то основная гипотеза отвергается.

In [4]:
(a - 2 * sigma, a + 2 * sigma)

(16.6, 17.4)

__5. Проводится статистический тест__

Для конкретной выборки 𝑋 считается значение 𝑇(𝑋) , и если оно принадлежит Ω𝑎𝑙𝑝ℎ𝑎 , то заключаем, что данные противоречат гипотезе 𝐻0 , и принимается гипотеза 𝐻1.

Гипотеза о том, что шарики для подшипников имеют средний диаметр 17 мм, при выбранном уровне значимости $\alpha = 0.025$ отвергается.

In [5]:
stats.norm.ppf(1 - 0.025, 17, 0.2)

17.39199279690801

__Задача 2__

Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочное среднее равно 80, а объём равен 256.

#### Найдем доверительный интервал согласно формуле: 
$$ P \left( \overline{X} - t_{1 - \alpha/2, \: n - 1} \cdot \dfrac{S}{\sqrt{n}} \leq M(X) \leq \overline{X} + t_{1 - \alpha/2, \: n - 1} \cdot \dfrac{S}{\sqrt{n}} \right) = 1 - \alpha. $$

Здесь $\overline{X} = 80$, $S = 16$, $n = 256$, $𝑡1−𝛼/2,𝑛−1 = 2$


In [6]:
std_z2 = 16
mean_z2 = 80
n_z2 = 256
print ('Доверительный интервал в рамках:', (mean_z2 - 2*std_z2/np.sqrt(n_z2), mean_z2 + 2*std_z2/np.sqrt(n_z2)))

Доверительный интервал в рамках: (78.0, 82.0)


__Задача 3__

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:

202, 203, 199, 197, 195, 201, 200, 204, 194, 190

Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%?

In [7]:
sample_z3 = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
m_z3 = sample_z3.mean()
n_z3 = 10
alpha_z3 = 0.01
sigma_z3 = np.sqrt(((sample_z3 - m_z3)**2).sum()/(n_z3-1))
print ('𝜎 =', sigma_z3)

𝜎 = 4.453463071962462


In [8]:
# Находим в таблице значение коэффициента Стьюдента
t_z3 = 3.2498

In [9]:
print ('Доверительный интервал в рамках:', (m - t_z3*sigma_z3/np.sqrt(n_z3), m_z3 + t_z3*sigma_z3/np.sqrt(n_z3)))

Доверительный интервал в рамках: (12.923278457308843, 203.07672154269116)


Доверительный интервал, построенный с учётом уровня значимости 1%, = (193.9, 203.1), включает значение предполагаемого среднего значения а = 200, утверждение продавца, что средний вес пачки = 200 гр, верно.